aips/aip-55.md

```diff
@@ -0,0 +1,97 @@
+---
+aip: 55
+title: Generalize Transaction Authentication and Support Arbitrary K-of-N MultiKey Accounts
+author: davidiw, hariria
+discussions-to:
+Status: Draft
+last-call-end-date (*optional): <mm/dd/yyyy the last date to leave feedbacks and reviews>
+type: Platform
+created: 10/16/2023
+updated (*optional): <mm/dd/yyyy>
+requires (*optional): N/A
+---
+
+# AIP-55 - Generalize Transaction Authentication and Support Arbitrary K-of-N MultiKey Accounts
+
+## Summary
+
+Transactions submitted to Aptos contain a `RawTransaction` and a `TransactionAuthenticator`. The `TransactionAuthenticator` authorizes the execution of the transaction by the set of senders or approvers of the accounts within the transaction. The `TransactionAuthenticator` contains a mixture of general purpose authenticators like fee payer and multiagent but also some very specific types for single sender transactions such as Ed25519, MultiEd25519, and Secp256k1. Thus adding a new cryptographic proof for authorizing a transaction requires a new `TransactionAuthenticator` and `AccountAuthenticator` variant to support both the single and multiparty transaction authenticaton.
+
```

Comment:
I never had time to understand this: why do we have both `TransactionAuthenticator` and `AccountAuthenticator`?

Reply:
TransactionAuthenticator